update dbt checks docs (#20914)
dagster-io/dagster: docs/content/integrations/dbt/reference.mdx

## Loading dbt tests as asset checks

<Note>
  <strong>Dagster asset checks are currently an experimental feature. </strong>{" "}
  To provide feedback, join our{" "}
  <a href="https://github.com/dagster-io/dagster/discussions/16527">

>     GitHub discussion

Reviewer: We should keep the note but change the contents.

1. Asset checks for dbt have been enabled by default, starting 1.7.
2. You need at least `dbt-core>=1.6` to use its full functionality.

We can remove the second bullet once `dbt-core==1.5` is deprecated at the end of April 2024.

> Dagster uses [dbt indirect selection](https://docs.getdbt.com/reference/global-configs/indirect-selection) to select dbt tests. By default, Dagster won't set `DBT_INDIRECT_SELECTION` so that the set of tests selected by Dagster is the same as the selected by dbt. When required, Dagster will override `DBT_INDIRECT_SELECTION` to `empty` in order to explicitly select dbt tests. Examples:

Reviewer 2: ```suggestion
Dagster uses [dbt indirect selection](https://docs.getdbt.com/reference/global-configs/indirect-selection) to select dbt tests. By default, Dagster won't set `DBT_INDIRECT_SELECTION` so that the set of tests selected by Dagster is the same as those selected by dbt. When required, Dagster will override `DBT_INDIRECT_SELECTION` to `empty` in order to explicitly select dbt tests. For example:
```

> - materializing dbt assets and excluding their asset checks
> - executing dbt asset checks without materializing their assets

Reviewer 2: ```suggestion
- Materializing dbt assets and excluding their asset checks
- Executing dbt asset checks without materializing their assets
```

### Singular tests

Dagster will load both generic and singular tests as asset checks. In the event that your singular test depends on multiple dbt models, you can use dbt metadata to specify which Dagster asset it should target. These fields can be filled in as they would be for the dbt [ref function](https://docs.getdbt.com/reference/dbt-jinja-functions/ref).

```yaml
meta:
  dagster:
    ref:
      name: ...
      version: ...
      package_name: ...
```

> This requires dbt version 1.6 or later.

Reviewer 2: Do we need to specify this is dbt Core? Also, what exactly requires dbt 1.6 - the ability to specify this metadata? Something else?

Author: cc @rexledesma - is "Manifest metadata includes project_name" the relevant bit from https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.6?

Reviewer: Yeah, that's the relevant bit.

If this metadata isn't provided, Dagster won't ingest the test as an asset check. It will still run the test and emit a <PyObject object="AssetObservation" /> events with the test results.

### Disabling asset checks

> You can disable modeling your dbt tests as asset checks. The tests will still run and will be emitted as <PyObject object="AssetObservation"/> events. To do so you'll need to define a <PyObject module="dagster_dbt" object="DagsterDbtTranslator" /> with <PyObject module="dagster_dbt" object="DagsterDbtTranslatorSettings" /> that have asset checks disable. The following example disables asset checks when using <PyObject module="dagster_dbt" object="dbt_assets" decorator/>:

Reviewer 2: ```suggestion
You can disable modeling your dbt tests as asset checks. The tests will still run and will be emitted as <PyObject object="AssetObservation"/> events. To do so you'll need to define a <PyObject module="dagster_dbt" object="DagsterDbtTranslator" /> with <PyObject module="dagster_dbt" object="DagsterDbtTranslatorSettings" /> that have asset checks disabled. The following example disables asset checks when using <PyObject module="dagster_dbt" object="dbt_assets" decorator/>:
```

## Loading dbt tests as asset checks

<Note>

>   <strong>Asset checks for dbt have been enabled by default, starting in Dagster 1.7.</strong><br/>

Reviewer: Let's use the `dagster-dbt` version here (0.23.0).
